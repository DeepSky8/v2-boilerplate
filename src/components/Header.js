import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux';



export const Header = ({ startLogout }) => (
    <header className='header'>
        <div className='content-container'>
            <div className='header__content'>
                <Link
                    to="/dashboard"
                    className='header__title'>
                    <h1>Boilerplate</h1>
                </Link>
                <button className='button button--link' name='logout' onClick={startLogout}>Logout</button>
            </div>
        </div>
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);

// <NavLink to="/help" className={({ isActive }) =>
//                 isActive ? 'is-active' : undefined
//             }>
//                 Help
//             </NavLink>

// <NavLink to="/" className={({ isActive }) =>
// isActive ? 'is-active' : undefined
// }>Login
// </NavLink>

// <NavLink to="/create" className={({ isActive }) =>
// isActive ? 'is-active' : undefined
// }>
// Create Expense
// </NavLink>