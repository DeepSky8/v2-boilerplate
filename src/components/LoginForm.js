import React from 'react';

export default class LoginForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            username: '',
            password: '',
            error: ''
        };
    };

    onUsernameChange = (e) => {
        const username = e.target.value;
        this.setState(() => ({ username }))
    }

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }))
    }


    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.username || !this.state.password) {
            this.setState(() => ({ error: 'Please enter a valid username and password' }))
        } else {
            this.setState(() => ({ error: '' }))
            this.props.onSubmit({
                username: this.state.username,
                password: this.state.password
            })
        }
    };

    render() {
        return (

            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <label htmlFor='username'>Username
                        <input
                            id='username'
                            type='text'
                            placeholder='Required'
                            autoFocus
                            value={this.state.username}
                            onChange={this.onUsernameChange} />
                    </label>
                    <label htmlFor='password'>Password
                        <input
                            id='password'
                            type='text'
                            placeholder='Required'
                            value={this.state.password}
                            onChange={this.onPasswordChange} />
                    </label>
                    <button>Log In</button>
                </form>
            </div>

        )
    }
}