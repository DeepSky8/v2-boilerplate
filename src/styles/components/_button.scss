.button {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

// .button:hover {
//     background-color: adjust-color($color: $parchment, $hue: 0, $saturation: 15%, $lightness: -15%, $alpha: 1);
//     color: black;
// }

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}