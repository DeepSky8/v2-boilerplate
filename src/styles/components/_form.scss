.form {
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: $m-size;
    }
}

.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
}


.label-placement {
    display: flex;
    flex-direction: column;
    // font-size: $font-size-large;
    // font-weight: 300;
}