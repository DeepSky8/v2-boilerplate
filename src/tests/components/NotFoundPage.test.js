import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { NotFoundPage } from '../../components/NotFoundPage';

test('should display the 404 page', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<NotFoundPage />)
    expect(renderer.getRenderOutput()).toMatchSnapshot();
})