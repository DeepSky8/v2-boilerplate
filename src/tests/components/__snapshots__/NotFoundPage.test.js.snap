// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the 404 page 1`] = `
<div>
  404 - 
  <Link
    to="/"
  >
    Go Home
  </Link>
</div>
`;
